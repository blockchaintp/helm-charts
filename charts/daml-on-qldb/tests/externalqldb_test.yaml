suite: External jdbbc for qldb kv store
templates:
  - statefulset.yaml
release:
  name: test
  namespace: testns
tests:
  - it: Construts a jdbcuri from external when create is false
    set:
      postgresqlqldb:
        create: false
        external:
          host: "externaltesthost"
          user: "externalusername"
          password: "exernalpassword"
          database: "externaldatabase"
    asserts:
      - matchSnapshot: {}
