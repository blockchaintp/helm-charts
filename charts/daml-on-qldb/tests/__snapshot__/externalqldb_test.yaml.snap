Construts a jdbcuri from external when create is false:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: test-daml-on-qldb
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: daml-on-qldb
        app.kubernetes.io/version: 1.0.0
        component: daml
        daml: test-daml-on-qldb
        helm.sh/chart: daml-on-qldb-0.0.13
      name: test-daml-on-qldb
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: test-daml-on-qldb
          app.kubernetes.io/instance: test
          app.kubernetes.io/name: daml-on-qldb
          component: daml
          daml: test-daml-on-qldb
      template:
        metadata:
          labels:
            app: test-daml-on-qldb
            app.kubernetes.io/instance: test
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: daml-on-qldb
            app.kubernetes.io/version: 1.0.0
            component: daml
            daml: test-daml-on-qldb
            helm.sh/chart: daml-on-qldb-0.0.13
        spec:
          affinity: null
          containers:
          - args:
            - |
              sleep 40
              /opt/daml-on-qldb/entrypoint.sh \
                --txlogstore "jdbc:postgresql://externaltesthost:5432/postgresqlqldb?user=externalusername&password=exernalpassword" \
                --createaws true  \
                --region us-east-1 \
                --ledger test \
                --participant server-jdbc-url="jdbc:postgresql://test-postgresqlidx:5432/postgresqlidx?user=postgresqlidx&password=postgresqlidx",participant-id=daml-on-qldb-participant,port=39000,address=0.0.0.0 \
                ;
            command:
            - bash
            - -xc
            env:
            - name: AWS_ACCESS_KEY_ID
              valueFrom:
                secretKeyRef:
                  key: aws_access_key_id
                  name: aws-secrets
            - name: AWS_SECRET_ACCESS_KEY
              valueFrom:
                secretKeyRef:
                  key: aws_secret_access_key
                  name: aws-secrets
            image: blockchaintp/daml-on-qldb:BTP2.1.0
            imagePullPolicy: IfNotPresent
            name: daml-on-qldb
            ports:
            - containerPort: 39000
              name: daml-ledger-api
            volumeMounts:
            - mountPath: /secrets/jwt
              name: jwt-secret
          serviceAccountName: test-daml-on-qldb
          volumes:
          - name: jwt-secret
            secret:
              secretName: test-daml-on-qldb-cert
